tests/zenko_tests/node_tests/backbeat/tests/ingestion/ingestionS3C.js

```diff
@@ -9,27 +9,41 @@ const scalityUtils = new IngestionUtility(scalityS3Client, ringS3Client);
 const ringS3CUtils = new IngestionUtility(ringS3Client);
 const ingestionSrcBucket = process.env.RING_S3C_INGESTION_SRC_BUCKET_NAME;
 const srcLocation = process.env.RING_S3C_BACKEND_SOURCE_LOCATION;
+const ingestionNonVersionedSrcBucket = process.env.RING_S3C_INGESTION_SRC_NON_VERSIONED_BUCKET_NAME;
+// Multiplying by 3 to account for the 3 types of objects: simple, zero-byte, and multipart
+const nonVersionedObjectCount = parseInt(process.env.RING_S3C_INGESTION_NON_VERSIONED_OBJECT_COUNT_PER_TYPE, 10) * 3;
+const srcNonVersionedLocation = process.env.RING_S3C_BACKEND_SOURCE_NON_VERSIONED_LOCATION;
 const location = srcLocation;
 let INGESTION_DEST_BUCKET;
 let KEY_PREFIX;
 let OBJ_KEY;
 
 describe('Ingesting existing data from RING S3C bucket', () => {
-    beforeEach(() => {
+    beforeEach(function () {
         INGESTION_DEST_BUCKET = `ingestion-${uuid()}`;
         KEY_PREFIX = `${ingestionSrcBucket}-${uuid()}`;
         OBJ_KEY = `${KEY_PREFIX}/object-to-ingest-${uuid()}`;
+        this.testState = {
+            s3cBucketToClean: ingestionSrcBucket,
+        };
     });
 
-    afterEach(done => async.series([
-        next => ringS3CUtils.deleteAllVersions(
-            ingestionSrcBucket,
-            null,
-            next,
-        ),
-        next => scalityUtils.waitUntilEmpty(INGESTION_DEST_BUCKET, next),
-        next => scalityUtils.deleteVersionedBucket(INGESTION_DEST_BUCKET, next),
-    ], done));
+    afterEach(function (done) {
+        async.series([
+            next => ringS3CUtils.deleteAllVersions(
```

Comment:
even if the location is created before, since we delete here we should also create the objects here : so the test may be (re)played multiple times...

Reply:
Putting the objects is not something we can do in this case. The whole reason behind putting the objects before the location gets created is to have a specific type of null version that doesn't have an internal version id, this type of object can only be created when the bucket is non versioned which cannot happen once versioning gets enabled on it.

I want to avoid having to create the location mid tests for flakiness reasons which is why i went with this approach.